src/status_im/ui/screens/events.cljs

```diff
@@ -108,21 +108,20 @@
 
 (re-frame/reg-fx
   :http-post
-  (fn [{:keys [action data success-event-creator failure-event-creator]}]
-    (utils/http-post action
-                     data
-                     #(re-frame/dispatch (success-event-creator %))
-                     #(re-frame/dispatch (failure-event-creator %)))))
-
-(defn- http-get [{:keys [url response-validator success-event-creator failure-event-creator]}]
-  (if response-validator
-    (utils/http-get url
-                    response-validator
-                    #(re-frame/dispatch (success-event-creator %))
-                    #(re-frame/dispatch (failure-event-creator %)))
-    (utils/http-get url
-                    #(re-frame/dispatch (success-event-creator %))
-                    #(re-frame/dispatch (failure-event-creator %)))))
+  (fn [{:keys [action data success-event-creator failure-event-creator timeout-ms]}]
+    (let [on-success #(re-frame/dispatch (success-event-creator %))
+          on-error   #(re-frame/dispatch (failure-event-creator %))]
+      (cond
+        timeout-ms (utils/http-post action data on-success on-error timeout-ms)
+        :default   (utils/http-post action data on-success on-error)))))
+
+(defn- http-get [{:keys [url response-validator success-event-creator failure-event-creator timeout-ms]}]
+  (let [on-success #(re-frame/dispatch (success-event-creator %))
+        on-error   #(re-frame/dispatch (failure-event-creator %))
+        opts       (cond-> {}
+                     response-validator (assoc :valid-response? response-validator)
```

Comment:
Could be simplified with a static map? Then nil values could be handled in `utils/http-get`